src/hera/workflows/protocol.py

```diff
@@ -37,6 +38,20 @@
 """`TWorkflow` is a union type collection of all the Hera workflow type objects that can manage contexts"""
 
 
+@runtime_checkable
+class Buildable(Protocol):
+    """This runtime protocol indicates that an object can built into a standalone serializable representation."""
+
+    def build(self) -> BaseModel:
+        """Produce a serializable model object."""
+
+    def to_dict(self) -> Any:
+        """Produce a raw python dict structure."""
+
+    def to_yaml(self, *args, **kwargs) -> str:
```

Comment:
I don't think we need to_yaml to be part of the protocol

Reply:
So my usecase is wholly based off being able to have the CLI be able to infer the ability to `to_yaml` something. whether it's a Workflow, some other theoretical hera-native top-level object that's not a worfklow subclass, or some externally defined resource.

I think `to_yaml` is the **minimum** protocol function I require. I included `to_dict` and `build` primarily because it seems likely that any top-level object will have this `build` step, and the current Workflow function chain path is `to_yaml -> to_dict -> build`.